File: backend/app/schemas/delta.py
from datetime import datetime

from pydantic import BaseModel


class DeltaCard(BaseModel):
    dossier_slug: str
    dossier_title: str
    new_events_count: int
    change_summary: str
    sentiment_shift: dict | None = None
    new_entities: list[str] = []
    predictions_tested: list[str] = []
    phase_change: bool = False
    velocity_change: str | None = None
    ripple_alerts: list[str] = []
    significance_score: float = 0.0
    last_checked: datetime | None = None


class DeltaResponse(BaseModel):
    cards: list[DeltaCard]
    sixty_second_summary: dict | None = None

File: frontend/src/types/delta.ts
export interface DeltaCard {
  dossier_slug: string;
  dossier_title: string;
  new_events_count: number;
  change_summary: string;
  sentiment_shift: Record<string, number> | null;
  new_entities: string[];
  predictions_tested: string[];
  phase_change: boolean;
  velocity_change: string | null;
  ripple_alerts: string[];
  significance_score: number;
  last_checked: string | null;
}

export interface DeltaResponse {
  cards: DeltaCard[];
  sixty_second_summary: {
    most_important_change: string;
    biggest_risk: string;
    actionable_insight: string;
  } | null;
}
